ts/Core/Chart/StockChart.ts

```diff
@@ -1120,35 +1202,141 @@ Axis.prototype.setCompare = function (
  *
+ * @param {boolean} [cumulative]
+ *        Either enable or disable Cumulative Sum mode.
+ *        Can be one of `false` (default) or `true`.
+ */
+Series.prototype.setCumulative = function (
+    cumulative?: boolean,
+    redraw?: boolean
+): void {
+    this.initCumulative(cumulative);
+
+    // Survive to export, #5485
+    this.options.cumulative = this.userOptions.cumulative = cumulative;
+
+    if (pick(redraw, true)) {
+        this.chart.redraw();
+    }
+};
+
+/**
  * @ignore
- * @function Highcharts.Point#tooltipFormatter
+ * @function Highcharts.Series#initCumulative
+ */
+Series.prototype.initCumulative = function (cumulative?: boolean): void {
+    // Set or unset the modifyValue method
+    this.modifyValue = cumulative ? function (
+        this: Series,
+        value?: number|null,
+        index?: number
+    ): (number|undefined) {
+        if (value === null) {
+            value = 0;
+        }
+
+        if (typeof value !== 'undefined' && typeof index !== 'undefined') {
+            const prevPoint = index > 0 ? this.points[index - 1] : null;
+
+            // Get the modified value
+            if (prevPoint && prevPoint.cumulativeSum) {
+                value = correctFloat(prevPoint.cumulativeSum + value);
+            }
+
+            // Record for tooltip etc.
+            const point = this.points[index];
+
+            if (point) {
+                point.cumulativeSum = value;
+            }
+
+            return value;
+        }
+
+        return 0;
+    } : null as any;
+
+    // Mark dirty
+    if (this.chart.hasRendered) {
+        this.isDirty = true;
+    }
+};
+
+/**
+ * @ignore
+ * @function Highcharts.Series#getCumulativeExtremes
  *
- * @param {string} pointFormat
+ * @param {Array} [activeYData]
+ *        An array cointaining all the points' y values in a visible range.
  */
-Point.prototype.tooltipFormatter = function (pointFormat: string): string {
-    const point = this;
-    const { numberFormatter } = point.series.chart;
+Series.prototype.getCumulativeExtremes = function (
+    this: Series,
+    activeYData: Array<number>
+): [number, number] {
+    let cumulativeDataMin = Infinity,
+        cumulativeDataMax = -Infinity;
 
-    pointFormat = pointFormat.replace(
-        '{point.change}',
-        ((point.change as any) > 0 ? '+' : '') + numberFormatter(
-            point.change as any,
-            pick(point.series.tooltipOptions.changeDecimals, 2)
-        )
-    );
+    activeYData.reduce((prev, cur): number => {
+        const sum = prev + cur;
 
-    return pointTooltipFormatter.apply(this, [pointFormat]);
+        cumulativeDataMin = Math.min(cumulativeDataMin, sum, prev);
+        cumulativeDataMax = Math.max(cumulativeDataMax, sum, prev);
+
+        return sum;
+    });
+
+    return [cumulativeDataMin, cumulativeDataMax];
+};
+
+/**
+ * Highcharts Stock only. Set the cumulative mode on all series
+ * belonging to an Y axis after render time.
```

Comment:
What do you mean by `after render time`?

Reply:
> What do you mean by after render time?
We should ask the author of the compare :) I think he meant that we don't have to define compare on the chart's init in the chart's options, but we can turn it on later - "after the chart has been initially rendered".